File: 프로그래머스/Level1/위클리챌린지/8주차_최소직사각형.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    int d1 = 0;
    int d2 = 0;
    
    for(int i=0; i<sizes.size(); i++){
        if(sizes[i][0]<sizes[i][1]){
            swap(sizes[i][0], sizes[i][1]);
        }
        
        d1 = max(d1, sizes[i][0]);
        d2 = max(d2, sizes[i][1]);
    }
    
    answer = d1*d2;
    
    return answer;
}

File: c++/프로그래머스/Level1/위클리챌린지/8주차_최소직사각형.py
def solution(sizes):
    answer = 0
    my_list = []
    
    for size in sizes :
        if size[1] > size[0] :
            size[0], size[1] = size[1], size[0]
        my_list.append(size)
    
    s1 = sorted(my_list, key=lambda x:x[0], reverse=True)
    s2 = sorted(my_list, key=lambda x:x[1], reverse=True)
    
    h = s1[0][0]
    w = s2[0][1]
    
    answer = h*w
    
    return answer